File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Orders/Requests/OrderRequest.cs
using ReiDoChopp.DataTransfer.OrderAdditionalFees.Request;
using ReiDoChopp.DataTransfer.OrdersProducts.Request;
using System;

namespace ReiDoChopp.DataTransfer.Orders.Request
{
    public class OrderRequest
    {
        public int? AttendantId { get; set; }
        public DateTime OrderDate { get; set; }
        public double? Discount { get; set; }
        public OrderAdditionalFeeRequest[] OrderAdditionalFees { get; set; }
        public OrderProductRequest[] OrderProducts { get; set; }
    }
}

File: rei-do-chopp-site/src/app/orders/models/requests/order.request.ts
import { OrderAdditionalFeeRequest } from "./order-additional-fee.request";
import { OrderProductRequest } from "./order-product.request";


export class OrderRequest {
  AttendantId?: number;
  OrderDate: Date;
  Discount?: number;
  OrderAdditionalFees: OrderAdditionalFeeRequest[];
  OrderProducts: OrderProductRequest[];

  constructor(params?: Partial<OrderRequest>) {
    this.AttendantId = params.AttendantId ?? null;
    this.OrderDate = params.OrderDate || null;
    this.Discount = params.Discount || null;
    this.OrderAdditionalFees = params.OrderAdditionalFees || [];
    this.OrderProducts = params.OrderProducts || [];
  }
}
